### PR TITLE
LiftAll for dependent type classes

### PR DESCRIPTION
**Pull request by Atry**
Note that this PR depends on #695

### CODE REVIEW

File: core/src/main/scala/shapeless/typeoperators.scala [hunk 1]
```diff
@@ -201,3 +201,34 @@ object Lub {
...
+  @macrocompat.bundle
+  private[WitnessThe] final class Macros(val c: whitebox.Context) {
+    import c.universe._
+    def apply[TypeClassConstraint: WeakTypeTag](typeClass: Tree): Tree = {
+      q"new _root_.shapeless.WitnessThe.WitnessTheAux[${weakTypeOf[TypeClassConstraint]}, ${typeClass.tpe}]($typeClass): _root_.shapeless.WitnessThe.Aux[${weakTypeOf[TypeClassConstraint]}, ${typeClass.tpe}]"
+    }
+  }
+
+  implicit def apply[TypeClassConstraint](implicit typeClass: TypeClassConstraint)
+    : WitnessThe.Aux[TypeClassConstraint, /* typeClass.type */ _ <: TypeClassConstraint] =
```
Comment (milessabin) [inline core/src/main/scala/shapeless/typeoperators.scala]: Can you motivate the wildcard here for me?

**Comment by milessabin** [comment]
Why is this a new type class rather than an adjustment to the instances for the existing `LiftAll`?

**Comment by milessabin** [comment]
We can move this to 2.4.0 in that case.

**Comment by joroKr21** [comment]
#695 was closed. Does mean this is no longer viable?

**Comment by milessabin** [comment]
@joroKr21 I was never really happy with the way #695 evolved and both it and this PR seem to have been abandoned. So I think close this one too, unless you or @Atry feel motivated to pick them up and get them over the finishing line.

**Comment by joroKr21** [comment]
Closing because the PR this depends on was not accepted.

**Pull request closed without merging**